hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -64,6 +65,13 @@ func createArmTypes(
 					astmodel.TypeDefinition{},
 					errors.Wrapf(err, "unable to create arm resource spec definition for resource %s", name)
 			}
+
+			if deffed, ok := kubeNameToArmDefs[kubeSpecName]; ok {
```

Comment:
minor: Include comment explaining that this is checking for conflicting `kubeNameToArmDefs`, which can happen ... somehow? Might be good including a few details.

Reply:
If it happens it's a bug elsewhere, really this should probably panic 🤔